### PR TITLE
Add Oracle Management Agent image

### PR DESCRIPTION
**Pull request by drupanchal**
Initial commit adding support to run Oracle Management Agent in a container. Refer to the Readme.md for instructions on how to validate the commit.

### CODE REVIEW

File: OracleManagementAgent/COPYRIGHT [hunk 1]
```diff
@@ -0,0 +1 @@
+Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleManagementAgent/COPYRIGHT:1]: This fine is not required.

File: OracleManagementAgent/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+The Universal Permissive License (UPL), Version 1.0
```
Comment (Djelibeybi) [inline OracleManagementAgent/LICENSE.txt:1]: This file is not required. We already have a `LICENSE.txt` in this repo.

File: OracleManagementAgent/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
+<!--
+    Copyright (c) 2022, Oracle and/or its affiliates.
+    Licensed under the Universal Permissive License v 1.0 as shown at
+    https://oss.oracle.com/licenses/upl.
+-->
+Oracle Management Agent on Docker
+=====
+This repository contains sample container configurations to facilitate installation and environment setup for DevOps users. This project includes Dockerfiles based on Oracle Linux and Oracle OpenJDK 8.
+
+The certification of Oracle Management Agent on Docker does not require the use of any file presented in this
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:10]: Please replace "on Docker" with "in a container"

File: OracleManagementAgent/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+repository. Customers and users are welcome to use them as starters, and customize/tweak, or create
+from scratch new scripts and Dockerfiles.
+
+
+## How to build and run
+
+Oracle Management Agent images use the Oracle Linux 7 Slim docker image as the base image.
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:17]: Please replace "docker" with "container"

File: OracleManagementAgent/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+* Latest Management Agent software for Linux (Linux-x86_64/latest/oracle.mgmt_agent.zip)
+
+#### Build steps
+
+1. [Download the Management Agent software](https://cloud.oracle.com/macs)
+
+    **Note: Select 'Downloads and Keys' then download 'Agent for LINUX (X86_64)' of the package type ZIP.**
+    ```shell
+    Copy-paste link: https://cloud.oracle.com/macs
+    ```
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:30]: This is not a shell command.

File: OracleManagementAgent/README.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    $ cp oracle.mgmt_agent.zip OracleManagementAgent/dockerfiles/latest/
+    ```
+
+1. Change directory to build the dockerfile
+
+    ```shell
+    $ cd OracleManagementAgent/dockerfiles/latest/
+    ```
+
+1. Build the Docker image
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:44]: Please rephrase this as "Use Docker to build a container image"

File: OracleManagementAgent/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    ```shell
+    $ docker build -t oracle/mgmtagent-container .
+    ```
+
+1. [Create and Download the Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html)
+
+    **Note: Follow the steps in the 'Create Install Key' and 'Configure a Response File' sections to create the install key and save it as input.rsp.**
+    ```shell
+    $ docker volume create mgmtagent-volume
+    ```
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:55]: This is a duplicate code block and should be removed.

File: OracleManagementAgent/README.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    # create any necessary dirs
+    $ mkdir -p /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret
+    $ cp input.rsp /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret/
+    ```
+
+1. [Apply Prerequisites for Deploying Management Agents](https://docs.oracle.com/en-us/iaas/management-agents/doc/perform-prerequisites-deploying-management-agents.html)
+
+    **Note: Without the prerequisites applied the agent may not function as expected.**
+
+1. Start the Docker container
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:79]: Please rephrase this as "Use Docker to start a container"

File: OracleManagementAgent/README.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+## License
+To download and run the Oracle Management Agent, regardless whether inside or outside a container, you must download the binaries from the Oracle website and accept the license indicated at that page.
+
+Oracle Linux is licensed under the [Oracle Linux End-User License Agreement](https://oss.oracle.com/ol/EULA).
+
+All scripts and files hosted in this project and GitHub [`docker-images/OracleManagementAgent`](./) repository, required to build the Docker images are, unless otherwise noted, released under the [UPL 1.0](https://oss.oracle.com/licenses/upl/) license.
+
+## Customer Support
+Oracle Management Agent docker image is supported for Oracle Linux 7. For more details please see My Oracle Support.
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:121]: Please replace "docker" with "container"

File: OracleManagementAgent/dockerfiles/build.sh [hunk 10]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/build.sh:2]: Please remove "All rights reserved." as we no longer include this phrase.

File: OracleManagementAgent/dockerfiles/latest/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/Dockerfile:1]: Please remove "All rights reserved." as we no longer include this phrase.

File: OracleManagementAgent/dockerfiles/latest/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+# Run:
+#      $ docker build -t oracle/mgmtagent-container .
+#
+# This command is already scripted in build.sh so you can alternatively run
+#		$ bash build.sh
+#
+
+FROM oraclelinux:7-slim as builder
+
+MAINTAINER Drupad Panchal <drupad.panchal@oracle.com>
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/Dockerfile:30]: The `MAINTAINER` directive is deprecated and should not be used.

File: OracleManagementAgent/dockerfiles/latest/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+# -----------------------
+# Put all downloaded files in the same directory as this Dockerfile
+# Run:
+#      $ docker build -t oracle/mgmtagent-container .
+#
+# This command is already scripted in build.sh so you can alternatively run
+#		$ bash build.sh
+#
+
+FROM oraclelinux:7-slim as builder
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/Dockerfile:28]: Please remove "as builder" as this is not a multistage build.

File: OracleManagementAgent/dockerfiles/latest/container-scripts/common.sh [hunk 14]
```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/container-scripts/common.sh:1]: Please remove "All rights reserved." as we no longer include this phrase.

File: OracleManagementAgent/dockerfiles/latest/container-scripts/install_zip.sh [hunk 15]
```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/container-scripts/install_zip.sh:1]: Please remove "All rights reserved." as we no longer include this phrase.

File: OracleManagementAgent/dockerfiles/latest/container-scripts/watchdog.sh [hunk 16]
```diff
@@ -0,0 +1,302 @@
+#!/bin/sh
+# Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/container-scripts/watchdog.sh:2]: Please remove "All rights reserved." as we no longer include this phrase.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
This PR is still in draft but I figured you could use this feedback while you're still working on it.

File: OracleManagementAgent/LICENSE.txt (hunk 2 above)
Comment (drupanchal) [reply OracleManagementAgent/LICENSE.txt]: removed

File: OracleManagementAgent/README.md (hunk 4 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: fixed

File: OracleManagementAgent/README.md [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+from scratch new scripts and Dockerfiles.
+
+
+## How to build and run
+
+Oracle Management Agent images use the Oracle Linux 7 Slim docker image as the base image.
+
+### Prerequisites
+
+* Latest Management Agent software for Linux (Linux-x86_64/latest/oracle.mgmt_agent.zip)
```
[the remark this answers is not in the document]
Comment (drupanchal) [reply OracleManagementAgent/README.md]: fixed

File: OracleManagementAgent/README.md (hunk 6 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: fixed

File: OracleManagementAgent/README.md (hunk 8 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: fixed

File: OracleManagementAgent/README.md [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+    $ docker run -d --name mgmtagent-container --hostname mgmtagent1 -v mgmtagent-volume:/opt/oracle:rw --restart unless-stopped oracle/mgmtagent-container:latest
+    ```
+    **Description of Docker run parameters used above**
+    ```shell
+    -d: Starts mgmtagent-container in detached mode
+    --name mgmtagent-container: The name given to the container to identify it.
+    --hostname mgmtagent1: Assign mgmtagent1 as the containers internal hostname. This can be any hostname compliant string and it will be used to identify the Management Agent instance in the OMC Console.
+    -v mgmtagent-volume:/opt/oracle:rw: Mounts the volume mgmtagent-volume created on host filesystem inside the container at /opt/oracle with Read/Write privileges.
+    --restart unless-stopped: Unless explicitly stopped, this restart policy restarts mgmtagent-container automatically when docker restarts.
+    ```
```
[the remark this answers is not in the document]
Comment (drupanchal) [reply OracleManagementAgent/README.md]: fixed

File: OracleManagementAgent/dockerfiles/latest/Dockerfile (hunk 12 above)
Comment (drupanchal) [reply OracleManagementAgent/dockerfiles/latest/Dockerfile]: fixed

File: OracleManagementAgent/dockerfiles/latest/container-scripts/common.sh (hunk 14 above)
Comment (drupanchal) [reply OracleManagementAgent/dockerfiles/latest/container-scripts/common.sh]: fixed

File: OracleManagementAgent/COPYRIGHT (hunk 1 above)
Comment (drupanchal) [reply OracleManagementAgent/COPYRIGHT]: removed

File: OracleManagementAgent/README.md (hunk 9 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: fixed

File: OracleManagementAgent/dockerfiles/latest/Dockerfile (hunk 11 above)
Comment (drupanchal) [reply OracleManagementAgent/dockerfiles/latest/Dockerfile]: fixed

File: OracleManagementAgent/dockerfiles/latest/container-scripts/install_zip.sh (hunk 15 above)
Comment (drupanchal) [reply OracleManagementAgent/dockerfiles/latest/container-scripts/install_zip.sh]: fixed

**Comment by drupanchal** [comment]
> This PR is still in draft but I figured you could use this feedback while you're still working on it.

Thank you for reviewing the draft and suggesting the fixes, I found them very helpful. I've applied the necessary fixes and submitted the PR for your final review.

File: OracleManagementAgent/README.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+
+    **Note: Select 'Downloads and Keys' then download 'Agent for LINUX (X86_64)' of the package type ZIP.**
+
+1. Copy the downloaded bundle to the same directory as the Dockerfile
+
+    ```shell
+    $ cp oracle.mgmt_agent.zip OracleManagementAgent/dockerfiles/latest/
+    ```
+
+1. Change directory to build the dockerfile
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:31]:
```suggestion
1. Change to the directory in which the `Dockerfile` for this container image is located
```

File: OracleManagementAgent/README.md [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+
+    ```shell
+    $ cd OracleManagementAgent/dockerfiles/latest/
+    ```
+
+1. Use Docker to build a container image
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:37]:
```suggestion
1. Build the container image
```

File: OracleManagementAgent/README.md [hunk 21]
```diff
@@ -0,0 +1,118 @@
...
+
+    ```shell
+    $ docker build -t oracle/mgmtagent-container .
+    ```
+
+1. [Create and Download the Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html)
+
+    **Note: Follow the steps in the 'Create Install Key' and 'Configure a Response File' sections to create the install key and save it as input.rsp.**
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:45]:
```suggestion
1. Follow the steps in the [Create Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/management-agents-administration-tasks.html#GUID-C841426A-2C32-4630-97B6-DF11F05D5712) and [Configure a Response File](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html#GUID-5D20D4A7-616C-49EC-A994-DA383D172486) sections of the [Management Agent](https://docs.oracle.com/en-us/iaas/management-agents/index.html) documentation to create an install key and save it locally as `input.rsp`.
```

File: OracleManagementAgent/README.md [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+        "Mountpoint": "/var/lib/docker/volumes/mgmtagent-volume/_data",
+    ```
+
+1. Copy the Install Key (input.rsp) into the shared Docker volume Mountpoint
+
+    ```shell
+    # create any necessary dirs
+    $ mkdir -p /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret
+    $ cp input.rsp /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret/
+    ```
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:63]: Is the `input.rsp` required for first run or every run? This is not best practice when it comes to sharing content into a container. You should either use secrets (if the `input.rsp` contains credentials) or use a first-run script that obtains those secrets and stores them in the container.

File: OracleManagementAgent/README.md [hunk 23]
```diff
@@ -0,0 +1,118 @@
...
+
+1. [Apply Prerequisites for Deploying Management Agents](https://docs.oracle.com/en-us/iaas/management-agents/doc/perform-prerequisites-deploying-management-agents.html)
+
+    **Note: Without the prerequisites applied the agent may not function as expected.**
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:67]:
```suggestion
1. Ensure your tenancy is configured correctly by [applying the documented prerequisites for deploying management agents](https://docs.oracle.com/en-us/iaas/management-agents/doc/perform-prerequisites-deploying-management-agents.html)
```

File: OracleManagementAgent/README.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+
+1. Use Docker to start a container
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:69]:
```suggestion
1. Start a container
```

File: OracleManagementAgent/README.md [hunk 25]
```diff
@@ -0,0 +1,118 @@
...
+    | --------- | ----------- |
+    | -d | Starts mgmtagent-container in detached mode |
+    | --name mgmtagent-container | The name given to the container to identify it. |
+    | --hostname mgmtagent1 | Assign mgmtagent1 as the containers internal hostname. This can be any hostname compliant string and it will be used to identify the Management Agent instance in the OMC Console. |
+    | -v mgmtagent-volume | /opt/oracle:rw: Mounts the volume mgmtagent-volume created on host filesystem inside the container at /opt/oracle with Read/Write privileges. |
+    | --restart unless-stopped | Unless explicitly stopped, this restart policy restarts mgmtagent-container automatically when docker restarts. |
+    <!-- markdownlint-enable MD033 -->
+
+
+#### Helpful Docker Administration commands
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:87]:
```suggestion
#### Helpful administration commands
```

File: OracleManagementAgent/README.md [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+
+1. Starting a stopped Management Agent Container
+
+    ```shell
+    $ docker container start mgmtagent-container
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:92]:
```suggestion
    $ docker start mgmtagent-container
```

File: OracleManagementAgent/README.md [hunk 27]
```diff
@@ -0,0 +1,118 @@
...
+    ```
+
+1. Stopping a running Management Agent Container
+
+    ```shell
+    $ docker container stop mgmtagent-container
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:98]:
```suggestion
    $ docker stop mgmtagent-container
```

File: OracleManagementAgent/README.md [hunk 28]
```diff
@@ -0,0 +1,118 @@
...
+    ```
+
+1. Inspecting logs of Management Agent Container
+
+    ```shell
+    $ docker container logs mgmtagent-container
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:104]:
```suggestion
    $ docker logs mgmtagent-container
```

File: OracleManagementAgent/README.md [hunk 29]
```diff
@@ -0,0 +1,118 @@
...
+
+## License
+To download and run the Oracle Management Agent, regardless whether inside or outside a container, you must download the binaries from the Oracle website and accept the license indicated at that page.
+
+Oracle Linux is licensed under the [Oracle Linux End-User License Agreement](https://oss.oracle.com/ol/EULA).
+
+All scripts and files hosted in this project and GitHub [`docker-images/OracleManagementAgent`](./) repository, required to build the Docker images are, unless otherwise noted, released under the [UPL 1.0](https://oss.oracle.com/licenses/upl/) license.
+
+## Customer Support
+Oracle Management Agent container image is supported for Oracle Linux 7. For more details please see My Oracle Support.
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:115]: I'm not sure I understand this line. Can I run this container only on Oracle Linux 7? Please provide a link to the specific My Oracle Support Article ID.

File: OracleManagementAgent/dockerfiles/latest/Dockerfile [hunk 30]
```diff
@@ -0,0 +1,55 @@
...
+      "issues"="https://github.com/oracle/docker-images/issues"         \
+      "volume.data"="/opt/oracle"
+
+# Install packages to install and run Management Agent
+RUN set -eux; \
+    yum -y install \
+    java-1.8.0-openjdk-devel.x86_64 \
+    sudo \
+    unzip \
+    && yum clean all
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/Dockerfile:41]:
```suggestion
    && rm -rf /var/cache/yum
```

File: OracleManagementAgent/dockerfiles/latest/Dockerfile [hunk 31]
```diff
@@ -0,0 +1,55 @@
...
+# Default to UTF-8 file.encoding
+ENV LANG en_US.UTF-8
+ENV LC_ALL=en_US.UTF-8
+
+# Copy container scripts to initialize the environment and start Management Agent
+COPY ./container-scripts/*.sh /opt/oracle-mgmtagent-bootstrap/scripts/
+RUN chmod 0544 /opt/oracle-mgmtagent-bootstrap/scripts/*.sh
+
+# Copy Management Agent Installer Zip bundle
+COPY ./oracle.mgmt_agent.zip /opt/oracle-mgmtagent-bootstrap/packages/
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/latest/Dockerfile:52]: How big is this zip file? If you're just going to unpack it, it would be better if it doesn't get copied into the image, but rather accessed at runtime so that it doesn't make the image larger.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
Please can you explain how the upgrader works and why you can't use a smaller bootstrap instead of storing an entire zip file in the image?

File: OracleManagementAgent/README.md (hunk 19 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 20 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 21 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 22 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md:85]: `input.rsp` is required up until the agent is registered. This is a runtime behavior dependent on what causes a registration failure, it can range from tenancy not configured correctly to any input misconfigured in the `input.rsp`. I take your point and have added docker-compose based container build and run execution. Docker-compose creates a secret and makes `input.rsp` available to the container without requiring additional user actions and hides the entire workflow.

File: OracleManagementAgent/README.md (hunk 23 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 24 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 25 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 26 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 28 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/dockerfiles/latest/Dockerfile (hunk 30 above)
Comment (drupanchal) [reply OracleManagementAgent/dockerfiles/latest/Dockerfile]: Fixed

File: OracleManagementAgent/README.md (hunk 27 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Fixed

File: OracleManagementAgent/README.md (hunk 29 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md]: Updated the doc and provided a link that better describes supported OS for Management Agent

File: OracleManagementAgent/dockerfiles/latest/Dockerfile (hunk 31 above)
Comment (drupanchal) [reply OracleManagementAgent/dockerfiles/latest/Dockerfile:52]: This is a great improvement suggestion. However, we add the zip to the container so that it benefits Kubernetes based container deployments so manual copying of the file to the persistent store is not required allowing deployments to be automated.

**Comment by drupanchal** [comment]
> Please can you explain how the upgrader works and why you can't use a smaller bootstrap instead of storing an entire zip file in the image?

Management Agent downloads the necessary bits from a non-public location to perform an upgrade and the scripts that are part of this PR simply orchestrate them and invoke the upgrader. At this time we do not have a separate smaller bootstrap responsible for installing/upgrading the Management Agent.

File: OracleManagementAgent/dockerfiles/latest/Dockerfile (hunk 31 above)
Comment (Djelibeybi) [reply OracleManagementAgent/dockerfiles/latest/Dockerfile:52]: Why add this to the image at all then? Have your `watchdog.sh` script test to see if the bootstrap package exists and if not, it can download and unpack it. That would make the image much smaller and it means that every time a new container is instantiated, it would automatically get the latest agent.

**Comment by Djelibeybi** [comment]
> At this time we do not have a separate smaller bootstrap responsible for installing/upgrading the Management Agent.

So don't put it in the image at all. The container is only useful when run on a host that has access to that non-public location, so install the bootstrap package on first run of the container instead.

File: OracleManagementAgent/README.md [hunk 32]
```diff
@@ -0,0 +1,136 @@
+<!--
+    Copyright (c) 2022, Oracle and/or its affiliates.
+    Licensed under the Universal Permissive License v 1.0 as shown at
+    https://oss.oracle.com/licenses/upl.
+-->
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:5]: Documentation files do not need copyright headers, so please remove this block.

File: OracleManagementAgent/README.md [hunk 33]
```diff
@@ -0,0 +1,136 @@
...
+Oracle Management Agent Container Image
+=====
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:7]:
```suggestion
# Oracle Management Agent Container Image

```

File: OracleManagementAgent/README.md [hunk 34]
```diff
@@ -0,0 +1,136 @@
...
+This repository contains sample container configurations to facilitate installation and environment setup for DevOps users. This project includes Dockerfiles based on Oracle Linux and Oracle OpenJDK 8.
+
+The certification of Oracle Management Agent in a container does not require the use of any file presented in this
+repository. Customers and users are welcome to use them as starters, and customize/tweak, or create
+from scratch new scripts and Dockerfiles.
+
+
+## How to build and run
+
+Oracle Management Agent images use the Oracle Linux 7 Slim container image as the base image.
```
Comment (Djelibeybi) [inline OracleManagementAgent/README.md:17]:
```suggestion
Oracle Management Agent images uses the official `oraclelinux:7-slim` container image as the base image. 
```

Comment (Djelibeybi) [reply OracleManagementAgent/README.md:17]: Though, I'm not entirely sure what the purpose of this sentence is. It doesn't provide any useful information that's not already contained in the `Dockerfile`.

File: OracleManagementAgent/dockerfiles/build.sh [hunk 35]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+# Copyright (c) 2022 Oracle and/or its affiliates.
+#
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+pushd ./latest
+docker build -t oracle/mgmtagent-container .
```
Comment (Djelibeybi) [inline OracleManagementAgent/dockerfiles/build.sh:7]: Why isn't this script checking to see if the files you need to build are in place before starting the build? I'm also not sure why you need this and a `docker-compose.yml`

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

**Comment by drupanchal** [comment]
> > At this time we do not have a separate smaller bootstrap responsible for installing/upgrading the Management Agent.
> 
> So don't put it in the image at all. The container is only useful when run on a host that has access to that non-public location, so install the bootstrap package on first run of the container instead.

The bootstrap ensures the Management Agent service always stays running so it must execute as the entrypoint. On the first run (or up until Management Agent service is fully installed and configured) the bootstrap retries several times to complete the install until successful. On successfully installing the agent it no longer attempts to execute that bootstrap code. So in a way it is setup to run as you describe.

File: OracleManagementAgent/README.md (hunk 32 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md:5]: Fixed

File: OracleManagementAgent/README.md (hunk 33 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md:7]: Fixed

File: OracleManagementAgent/README.md (hunk 34 above)
Comment (drupanchal) [reply OracleManagementAgent/README.md:17]: It's just an info statement that we found other docker-images projects had in their doc so we kept it thinking it was a convention

**Review by Djelibeybi — APPROVED** [review]
